include/settings/settings.h

```diff
@@ -104,15 +119,16 @@ struct settings_handler {
  */
 int settings_subsys_init(void);
 
+#if defined(CONFIG_SETTINGS_RAM_HANDLERS) /*  */
```

Comment:
ditto, for functions it should be fine to always have it available in the header since the build will fail when linking.

Reply:
OK, will change.

Reply:
And the doc will also fail (like it's failing now) otherwise.